### ISSUE TITLE
Add option to order elements inside a grid

### ISSUE THREAD

**Issue by romaninsh** [atk4/ui#405]
Currently you can take a Grid and make elements selectable with a relative ease:

http://agile-ui.readthedocs.io/en/latest/grid.html?highlight=grid#selection

It should be possible to produce grid order in a similar way.

1. Add a dragging handle for grid lines.
2. Provide an expression to fetch grid order `$ord->getOrder()`

This should return list of IDs: 392,29,493 which can either be stored in a hidden form field or communicated to the backend through AJAX, that is outside of this ticket's scope.

**Comment by romaninsh** [comment]
not sure if this is fully possible.

http://rubaxa.github.io/Sortable/

**Comment by ibelar** [comment]
Believe this one is more up to date and modern. 
https://github.com/Shopify/draggable

Here is a quick sample using grid:
http://jsbin.com/qoxivuj/edit?output

**Comment by romaninsh** [comment]
Yes, this looks perfect. Just need to have an event and dump all IDs into a field. I think there are 2 use-cases:

## 1. After drag finished, send AJAX request, e.g.:

``` php
$order = $grid->makeOrderable();
$order->onReorder(function($ids) use ($order){
    // to whatever we want
 });
```

## 2. reference order directly.

This can be a prompt to event:

``` php
$order->onReorder($field->jsInput()->set($order->jsOrder());
```

or any other way:

``` php
$button->on('click', new jsExpression('alert([])', [$order->jsOrder()]));
```

**Comment by ibelar** [comment]
This was implemented in #407 
Should probably be closed unless we need more feature
